include/nlohmann/detail/iterators/iteration_proxy.hpp

```diff
@@ -12,10 +12,13 @@ namespace nlohmann
 {
 namespace detail
 {
-template<typename string_type>
+template<typename string_type, typename std::enable_if<
```

Comment:
You can remove the `std::enable_if` part now. We now just assume that `to_string` will do the job of converting a `std::size_t` to `string_type`.